airbyte-integrations/connectors/destination-aws-datalake/destination_aws_datalake/destination.py

```diff
@@ -89,23 +123,28 @@ def check(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> AirbyteConn
             logger.error(f"""Could not create session on {connector_config.aws_account_id} Exception: {repr(e)}""")
             message = f"""Could not authenticate using {connector_config.credentials_type} on Account {connector_config.aws_account_id} Exception: {repr(e)}"""
             return AirbyteConnectionStatus(status=Status.FAILED, message=message)
+        except InvalidRegionError:
+            message = f"{connector_config.region} is not a valid AWS region"
+            logger.error(message)
+            return AirbyteConnectionStatus(status=Status.FAILED, message=message)
 
         try:
             aws_handler.head_bucket()
         except ClientError as e:
             message = f"""Could not find bucket {connector_config.bucket_name} in aws://{connector_config.aws_account_id}:{connector_config.region} Exception: {repr(e)}"""
             return AirbyteConnectionStatus(status=Status.FAILED, message=message)
 
-        with LakeformationTransaction(aws_handler) as tx:
-            table_location = "s3://" + connector_config.bucket_name + "/" + connector_config.bucket_prefix + "/" + "airbyte_test/"
-            table = aws_handler.get_table(
-                txid=tx.txid,
-                database_name=connector_config.lakeformation_database_name,
-                table_name="airbyte_test",
-                location=table_location,
-            )
-        if table is None:
-            message = f"Could not create a table in database {connector_config.lakeformation_database_name}"
+        tbl = "airbyte_test"
```

Comment:
this table name should be randomized and `check` should also attempt to delete this table before exiting. Right now, subsequent test runs fail with this error:
```
	 ERROR    airbyte:destination.py:147 Could not create table airbyte_*** in database airbyte-integration: AlreadyExistsException('An error occurred (AlreadyExistsException) when calling the CreateTable operation: Table already exists.')
	 =========================== short *** summary info ============================
	 FAILED integration_***s/integration_***.py::***_check_valid_config - Asser...
	 ========================= 1 failed, 3 passed in 48.70s =========================

```

Reply:
Hmm weird, `aws_handler.reset_table(db, tbl)` a couple of lines below this one should delete the table if it exists and all it's underlying data in s3 before it tries to create it again. Does your configured iam role or user have permissions to access/delete the table? The integration test logs show the following sequence of events:
1. A call made to GetTable returning "Table not found" (user may not have permissions to access it)
2. A call made to CreateTable returning "Table already exists" (error could be caused by table existing but user not being able to access it)
3. A call made to DeleteTable returning "Table not found"
Makes me think that for some reason the role or user does not have permissions to access the existing table? But I may be overlooking something? I'll make sure to randomize the table name in my next commit

Reply:
A scenario that the integration test should account for is that a previous run did not properly cleanup (it could be cancelled, interrupted, failed due to a hosting problem, CI problem or anything else).

Another problem is that `CHECK` should not assume that `airbyte_test` table does not exist in the destination, because a user could be using multiple connections to write to the same destination. It is safer to add a random suffix/preffix to the tables created by `CHECK` method to avoid a name collision.

Reply:
That makes sense!